src/commodore/util.ts

```diff
@@ -72,3 +78,98 @@ export function parseFileName(
 
   return facts;
 }
+
+function isObject(item: any) {
+  return item && typeof item === 'object' && !Array.isArray(item);
+}
+
+// Adapted from https://stackoverflow.com/a/37164538
+export function mergeConfig(base: any, config: any): any {
+  let output = clone(base);
+  if (isObject(base) && isObject(config)) {
+    Object.keys(config).forEach((key) => {
+      if (isObject(config[key])) {
+        if (!(key in base)) Object.assign(output, { [key]: config[key] });
+        else output[key] = mergeConfig(base[key], config[key]);
+      } else if (Array.isArray(config[key])) {
+        if (!(key in base)) {
+          Object.assign(output, { [key]: Array.from(config[key]) });
+        } else if (Array.isArray(output[key])) {
+          output[key].push(...config[key]);
+        } else {
+        }
+      } else {
+        Object.assign(output, { [key]: config[key] });
+      }
+    });
+  }
+  return output;
+}
+
+export var globalRepos: Map<string, RepoConfig> = new Map();
+
+export function globalRepoDir(globalRepoURL: string): string {
+  let repoUrl = new URL(globalRepoURL);
+  let repoPathInfo = parse(repoUrl.pathname);
+  let repoPath = `${repoUrl.host}${repoPathInfo.dir}/${repoPathInfo.name}`;
+  return cacheDir() + `/global-repos/${repoPath}`;
+}
+
+export async function cloneGlobalRepo(config: any): Promise<RepoConfig> {
```

Comment:
We only actually need the repo URL. This would also give use some type safety and not explode if `globalRepoURL` is missing or not a string.